Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -0,0 +1,801 @@
+  int nTracks = cacheTrk.size(), nTOFCls = cacheTOF.size();
+  LOG(INFO) << "Matching sector " << sec << ": number of tracks: " << nTracks << ", number of TOF clusters: " << nTOFCls;
+  if (!nTracks || !nTOFCls) {
+    return;
+  }
+  int itof0 = 0; // starting index in TOF clusters for matching of the track
+  int   detId[2][5]; // at maximum one track can fall in 2 strips during the propagation; the second dimention of the array is the TOF det index
+  float deltaPos[2][3]; // at maximum one track can fall in 2 strips during the propagation; the second dimention of the array is the residuals
+  int nStepsInsideSameStrip[2] = {0,0}; // number of propagation steps in the same strip (since we have maximum 2 strips, it has dimention = 2)
+  float deltaPosTemp[3];
+  std::array<float, 3> pos;
+  std::array<float, 3> posBeforeProp;
+  float posFloat[3]; 
+
+  LOG(DEBUG) << "Trying to match %d tracks" << cacheTrk.size();
+  for (int itrk = 0; itrk < cacheTrk.size(); itrk++) {
+    for (int ii = 0; ii < 2; ii++) {
+      detId[ii][2] = -1; // before trying to match, we need to inizialize the detId corresponding to the strip number to -1; this is the array that we will use to save the det id of the maximum 2 strips matched 
+      nStepsInsideSameStrip[ii] = 0;
+    }
+    int nStripsCrossedInPropagation = 0; // how many strips were hit during the propagation
+    auto& trefTrk = mTracksWork[cacheTrk[itrk]];
+    float minTrkTime = (trefTrk.getTimeMUS().getTimeStamp() - mSigmaTimeCut*trefTrk.getTimeMUS().getTimeStampError())*1.E6; // minimum time in ps
+    float maxTrkTime = (trefTrk.getTimeMUS().getTimeStamp() + mSigmaTimeCut*trefTrk.getTimeMUS().getTimeStampError())*1.E6; // maximum time in ps
+    int istep = 1; // number of steps
+    float step = 0.1; // step size in cm
+    //uncomment for local debug
+    /*
+    //trefTrk.getXYZGlo(posBeforeProp);
+    //float posBeforeProp[3] = {trefTrk.getX(), trefTrk.getY(), trefTrk.getZ()}; // in local ref system
+    //printf("Global coordinates: posBeforeProp[0] = %f, posBeforeProp[1] = %f, posBeforeProp[2] = %f\n", posBeforeProp[0], posBeforeProp[1], posBeforeProp[2]);
+    //Printf("Radius xy = %f", TMath::Sqrt(posBeforeProp[0]*posBeforeProp[0] + posBeforeProp[1]*posBeforeProp[1]));
+    //Printf("Radius xyz = %f", TMath::Sqrt(posBeforeProp[0]*posBeforeProp[0] + posBeforeProp[1]*posBeforeProp[1] + posBeforeProp[2]*posBeforeProp[2]));
+    */
+
+#ifdef _ALLOW_DEBUG_TREES_
+    (*mDBGOut) << "propOK"
+	       << "track=" << trefTrk << "\n";
+#endif
+
+    // initializing
+    for (int ii = 0; ii < 2; ii++){
+      for (int iii = 0; iii < 5; iii++){
+	detId[ii][iii] = -1;
+      }
+    }
+    int detIdTemp[5] = {-1, -1, -1, -1, -1}; // TOF detector id at the current propagation point
+    while (propagateToRefX(trefTrk, mXRef+istep*step, step) && nStripsCrossedInPropagation <=2 && mXRef+istep*step < Geo::RMAX){
```

Comment:
we are going to use TrackPar for fast propagation.
I did't understand why TrackParCov::rotate method is not present in the mother class TrackPar, while I found TrackPar::rotatePar method. Since TrackParCov inherits from TrackPar I exepected the same name and the use of virtualization.

Reply:
I see. Thanks

Reply:
Hello Ruben,

We committed the modification. Now we use TrackPar until the inner radius of TOF. You suggested to use this approach till we mathched a strip, we can also do that if you think it is important. For now, please, have a look at how we propagate. 
Note that:
- we consider only the z component of the B field during the first "course" propagation;
- **as before**, the step of the propagation till 371 cm is 2 cm, only inside TOF it is 0.1 cm.

Reply:
Thanks @chiarazampolli , I added one more comment, but given that this small-step propagation may / pad_search  may be the most expensive operation, we better merge it is it is, and optimize it later, when we the outer param + length integration + propagator with mat.budget LUT is available.

Reply:
Ok, I resolve here now, we have the other comment open, in case you change your mind and prefer this changes now.